src/ocm/components/clusterConfiguration/networkConfiguration/AdvancedNetworkFields.tsx

```diff
@@ -0,0 +1,135 @@
+    parseInt(
+      ((values.clusterNetworks && values.clusterNetworks[index].cidr) || '').split('/')[1],
+    ) || 1;
+
+  const formatClusterNetworkHostPrefix = (
+    e: React.ChangeEvent<HTMLInputElement>,
+    index: number,
+  ) => {
+    if (isNaN(parseInt(e.target.value))) {
+      setFieldValue(`clusterNetworks.${index}.hostPrefix`, clusterNetworkCidrPrefix(index));
+    }
+  };
+
+  const isSubnetIPv6 = (index: number) => (isDualStack ? !!index : false);
+
+  const clusterNetworkHostPrefixHelperText = (index: number) =>
+    isSubnetIPv6(index) ? IPv6PrefixHelperText : IPv4PrefixHelperText;
+
+  return (
+    <Grid hasGutter>
+      <FieldArray name="clusterNetworks">
+        {() => (
+          <FormGroup fieldId="clusterNetworks" labelInfo={isDualStack && 'Primary'}>
+            {values.clusterNetworks?.map((_, index) => {
+              return (
+                <StackItem key={index} className={'network-field-group'}>
+                  <InputField
+                    name={`clusterNetworks.${index}.cidr`}
+                    label="Cluster network CIDR"
+                    helperText={clusterCidrHelperText}
+                    isRequired
+                    labelInfo={index == 0 && values.stackType == DUAL_STACK ? 'Primary' : ''}
+                  />
+                  <InputField
+                    name={`clusterNetworks.${index}.hostPrefix`}
+                    label="Cluster network host prefix"
+                    type={TextInputTypes.number}
+                    min={clusterNetworkCidrPrefix(index)}
+                    max={
+                      isSubnetIPv6(index)
+                        ? PREFIX_MAX_RESTRICTION.IPv6
+                        : PREFIX_MAX_RESTRICTION.IPv4
+                    }
+                    onBlur={(e) =>
+                      formatClusterNetworkHostPrefix(
+                        e as React.ChangeEvent<HTMLInputElement>,
+                        index,
+                      )
+                    }
+                    helperText={clusterNetworkHostPrefixHelperText(index)}
+                    isRequired
+                  />
+                </StackItem>
+              );
+            })}
+          </FormGroup>
+        )}
+      </FieldArray>
+
+      {typeof errors.clusterNetworks === 'string' && (
+        <Alert variant={AlertVariant.warning} title={errors.clusterNetworks} isInline />
+      )}
+
+      <FieldArray name="serviceNetworks">
+        {() => (
+          <FormGroup fieldId="serviceNetworks" labelInfo={isDualStack && 'Primary'}>
+            {values.serviceNetworks?.map((_, index) => (
+              <StackItem key={index} className={'network-field-group'}>
+                <InputField
+                  name={`serviceNetworks.${index}.cidr`}
+                  label="Service network CIDR"
+                  helperText={serviceCidrHelperText}
+                  isRequired
+                  labelInfo={index == 0 && isDualStack ? 'Primary' : ''}
```

Comment:
```suggestion
                  labelInfo={index === 0 && isDualStack ? 'Primary' : ''}
```

Reply:
We could enhance our  `eslint` config to highlight this type of issues, and have them fixed automatically.
https://eslint.org/docs/rules/eqeqeq

Maybe we could create a separate ticket, as there could be other interesting rules to add, etc.

Reply:
totally agree https://github.com/openshift-assisted/assisted-ui-lib/pull/1175#pullrequestreview-935990044 :) Im already working on it